src/main/java/ch/njol/skript/expressions/ExprAllSignLines.java

```diff
@@ -0,0 +1,190 @@
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.Aliases;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.effects.Delay;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.Block;
+import org.bukkit.block.Sign;
+import org.bukkit.event.Event;
+import org.bukkit.event.block.SignChangeEvent;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Name("All Sign Lines")
+@Description("All lines of a sign. Can be changed, but remember that there is a 16 character limit per line (including colour codes that use 2 characters each).")
+@Examples({"on rightclick on sign:",
+	"	all lines of clicked sign contains \"[Kill Me]\":",
+	"		kill the player",
+	"	set line 3 to \"%player%\""})
+@Since("INSERT VERSION")
+public class ExprAllSignLines extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprAllSignLines.class, String.class, ExpressionType.PROPERTY, "[the] [all] lines [of %blocks%]");
+	}
+
+	private static final ItemType sign = Aliases.javaItemType("sign");
+
+	@SuppressWarnings("null")
+	private Expression<Block> blocks;
+	private boolean multiple = false;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		blocks = (Expression<Block>) exprs[exprs.length - 1];
+		multiple = !blocks.isSingle();
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected String[] get(Event e) {
+		Block[] blocks = this.blocks.getArray(e);
+		if (blocks.length == 0)
+			return new String[0];
+
+		List<String> lines = new ArrayList<>();
+
+		boolean isSignChangeEvent = e instanceof SignChangeEvent;
+		if (getTime() >= 0 && this.blocks.isDefault() && isSignChangeEvent && !Delay.isDelayed(e)) {
+			SignChangeEvent event = (SignChangeEvent) e;
+			lines.addAll(List.of(event.getLines()));
+		}
+
+		if (!multiple && (isSignChangeEvent && blocks[0].equals(((SignChangeEvent) e).getBlock())))
+			return lines.toArray(new String[0]);
+
+		for (Block block : blocks) {
+			if (!sign.isOfType(block))
+				continue;
+
+			Sign sign = (Sign) block.getState();
+			lines.addAll(Arrays.asList(sign.getLines()));
+		}
+
+		return lines.toArray(new String[0]);
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.DELETE || mode == ChangeMode.SET)
+			return new Class[]{String.class};
+		return null;
+	}
+
+	static boolean hasUpdateBooleanBoolean = true;
```

Comment:
We could have a better field name other than hasUpdateBooleanBoolean, like maybe hasForceUpdate.